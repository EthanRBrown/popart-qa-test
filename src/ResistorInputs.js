import React from 'react'
import { FormGroup, Label, Input } from 'reactstrap'
import { Col } from 'reactstrap'
import { Button } from 'reactstrap'
import ColHeader from './ColHeader'

const ResistorInputs = props => 
	<div>
		<ColHeader title="Inputs" />
		<div>
			<FormGroup row>
				<Label sm={2}>R<sub>equiv</sub></Label>
				<Col sm={10}>
					<Input type="number" disabled value={props.requiv} />
				</Col>
			</FormGroup>
			<FormGroup row>
				<Label sm={2}>R<sub>1</sub></Label>
				<Col sm={10}>
					<Input type="number" name="R1" onChange={evt => props.setR1(evt.target.value)} value={props.r1} />
				</Col>
			</FormGroup>
			<FormGroup row>
				<Label sm={2}>R<sub>2</sub></Label>
				<Col sm={10}>
					<Input type="number" name="R2" onChange={evt => props.setR2(evt.target.value)} value={props.r2} />
				</Col>
			</FormGroup>
			<FormGroup>
				<Button color="primary" onClick={props.showSeekDialog}>Seek</Button>
			</FormGroup>
		</div>
	</div>

export default ResistorInputs
