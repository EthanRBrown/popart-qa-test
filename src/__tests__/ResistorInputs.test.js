import React from 'react';
import renderer from 'react-test-renderer'
import ResistorInputs from '../ResistorInputs.js'

it('renders resistance values correctly', () => {
  const comp = renderer.create(<ResistorInputs r1={100} r2={100} requiv={50} />)
  expect(comp).toMatchSnapshot()
})
