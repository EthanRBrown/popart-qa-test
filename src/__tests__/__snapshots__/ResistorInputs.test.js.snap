// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders resistance values correctly 1`] = `
<div>
  <div>
    <h4>
      Inputs
    </h4>
    <hr />
  </div>
  <div>
    <div
      className="row form-group"
    >
      <label
        className="col-sm-2 col-form-label"
        htmlFor={undefined}
      >
        R
        <sub>
          equiv
        </sub>
      </label>
      <div
        className="col col-sm-10"
      >
        <input
          className="form-control"
          disabled={true}
          type="number"
          value={50}
        />
      </div>
    </div>
    <div
      className="row form-group"
    >
      <label
        className="col-sm-2 col-form-label"
        htmlFor={undefined}
      >
        R
        <sub>
          1
        </sub>
      </label>
      <div
        className="col col-sm-10"
      >
        <input
          className="form-control"
          name="R1"
          onChange={[Function]}
          type="number"
          value={100}
        />
      </div>
    </div>
    <div
      className="row form-group"
    >
      <label
        className="col-sm-2 col-form-label"
        htmlFor={undefined}
      >
        R
        <sub>
          2
        </sub>
      </label>
      <div
        className="col col-sm-10"
      >
        <input
          className="form-control"
          name="R2"
          onChange={[Function]}
          type="number"
          value={100}
        />
      </div>
    </div>
    <div
      className="form-group"
    >
      <button
        className="btn btn-primary"
        onClick={[Function]}
        type={undefined}
      >
        Seek
      </button>
    </div>
  </div>
</div>
`;
